File: Data Structures & Algorithms/string-encode-and-decode/submission-2.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        res=""
        for s in strs:
            res += s + '/'
        return res

    def decode(self, s: str) -> List[str]:
        res=[]
        data=""
        for c in s:
            if c == '/':
                res.append(data)
                data=""
            else:
                data+=c
        return res

File: Data Structures & Algorithms/anagram-groups/submission-2.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>>data;
        for(auto s:strs)
        {
            string sortedres=s;
            sort(sortedres.begin(),sortedres.end());
            data[sortedres].push_back(s);
        }
        vector<vector<string>>result;
        for(auto pair:data)
        {
            result.push_back(pair.second);
        }
        return result;

    }
};
